store/auth.go

```diff
@@ -173,7 +176,9 @@ func decodeUnauthorizedError(body io.ReadCloser) error {
 	var msg authError
 	dec := json.NewDecoder(body)
 	if err := dec.Decode(&msg); err != nil {
-		return fmt.Errorf("store authorization failed with unknown error: %v", err)
+		b, err := ioutil.ReadAll(body)
```

Comment:
I don't think/I'm not sure you can read the body again this way, it will be consumed at this point